packages/core/src/logs/index.ts

```diff
@@ -0,0 +1,144 @@
+ *
+ * @param key - The key of the log attribute
+ * @param value - The value of the log attribute
+ * @returns The serialized log attribute
+ */
+export function logAttributeToSerializedLogAttribute(key: string, value: unknown): SerializedLogAttribute {
+  switch (typeof value) {
+    case 'number':
+      return {
+        key,
+        value: { doubleValue: value },
+      };
+    case 'boolean':
+      return {
+        key,
+        value: { boolValue: value },
+      };
+    case 'string':
+      return {
+        key,
+        value: { stringValue: value },
+      };
+    default:
+      return {
+        key,
+        value: { stringValue: JSON.stringify(value) ?? '' },
+      };
+  }
+}
+
+/**
+ * Captures a log event and sends it to Sentry.
+ *
+ * @param log The log event to capture.
+ *
+ * @experimental This method will experience breaking changes. This is not yet part of
+ * the stable Sentry SDK API and can be changed or removed without warning.
+ */
+export function captureLog(
+  { level, message, attributes, severityNumber }: Log,
+  currentScope = getCurrentScope(),
+  client = getClient(),
+): void {
+  if (!client) {
+    DEBUG_BUILD && logger.warn('No client available to capture log.');
+    return;
+  }
+
+  const { _experiments, release, environment } = client.getOptions();
+  if (!_experiments?.enableLogs) {
+    DEBUG_BUILD && logger.warn('logging option not enabled, log will not be captured.');
+    return;
+  }
+
+  const [, traceContext] = _getTraceInfoFromScope(client, currentScope);
+
+  const logAttributes = {
+    ...attributes,
+  };
+
+  if (release) {
+    logAttributes.release = release;
+  }
+
+  if (environment) {
+    logAttributes.environment = environment;
+  }
+
+  const span = _getSpanForScope(currentScope);
+  if (span) {
+    // Add the parent span ID to the log attributes for trace context
+    logAttributes['sentry.trace.parent_span_id'] = span.spanContext().spanId;
+  }
+
+  const serializedLog: SerializedOtelLog = {
+    severityText: level,
+    body: {
+      stringValue: message,
+    },
+    attributes: Object.entries(logAttributes).map(([key, value]) => logAttributeToSerializedLogAttribute(key, value)),
+    timeUnixNano: `${new Date().getTime().toString()}000000`,
+    traceId: traceContext?.trace_id,
+    severityNumber: severityNumber ?? SEVERITY_TEXT_TO_SEVERITY_NUMBER[level],
+  };
+
+  const logBuffer = CLIENT_TO_LOG_BUFFER_MAP.get(client);
+  if (logBuffer === undefined) {
+    CLIENT_TO_LOG_BUFFER_MAP.set(client, [serializedLog]);
+    // Every time we initialize a new log buffer, we start a new interval to flush the buffer
+    return;
+  }
+
+  logBuffer.push(serializedLog);
+  if (logBuffer.length > MAX_LOG_BUFFER_SIZE) {
+    _INTERNAL_flushLogsBuffer(client, logBuffer);
+  }
+}
+
+/**
+ * Flushes the logs buffer to Sentry.
+ */
+export function _INTERNAL_flushLogsBuffer(client: Client, maybeLogBuffer?: Array<SerializedOtelLog>): void {
```

Comment:
The only thing that is not ideal about this approach is that we have to expose this method to allow us to flush. This is as we need a way for the browser and server-runtime-client to get access to the logs buffer.

Reply:
That still has bundle size hit because we need to add that stuff to the client. I'll think about this though, it's a good point.